GitCommands/Git/GitModule.cs

```diff
@@ -3141,7 +3145,7 @@ public string FormatBranchName([NotNull] string branchName)
                 throw new ArgumentNullException(nameof(branchName));
 
             string fullBranchName = GitCommandHelpers.GetFullBranchName(branchName);
-            if (String.IsNullOrEmpty(RevParse(fullBranchName)))
+            if (RevParse(fullBranchName) == null)
```

Comment:
`RevParse` returns `null` if no hash exists. The 'nil' value for `ObjectId` is `null`. I originally made `ObjectId` a struct, but the utility of `null` outweighed the performance benefit.